### ISSUE TITLE
Use of DataLoaders

### ISSUE THREAD

**Issue by danielkelshaw** [HamzaSardar/PyCoarseNet#5]
At the moment the data is just being sliced from an input array. This can work while the data is small but it would be much more convenient to use a `torch.utils.data.DataLoader`. This would allow you to run the model through in batches and gives functionality such as batch-wise shuffling. It is also more 'torch-thonic'.

https://github.com/HamzaSardar/PyCoarseNet/blob/47baebc46cdd1bf1b6ce11ff8abcdabf9e6129fc/pycoarsenet/training/model_setup.py#L36-L37
